modules/rest/src/main/java/eu/cloudnetservice/modules/rest/commands/RestCommand.java

```diff
@@ -0,0 +1,120 @@
+  private static final ConsoleSetupAnimation ANIMATION = new ConsoleSetupAnimation(HEADER, FOOTER, "&r> &e");
+  private final Console console;
+
+  private final V2HttpHandlerQRCode v2HttpHandlerQRCode;
+
+  public RestCommand(@NonNull Console console, @NonNull V2HttpHandlerQRCode v2HttpHandlerQRCode) {
+    this.console = console;
+    this.v2HttpHandlerQRCode = v2HttpHandlerQRCode;
+  }
+
+  @CommandPermission("cloudnet.command.rest.setup")
+  @CommandDescription("Goes trough the rest setup process")
+  @CommandMethod(value = "rest setup", requiredSender = ConsoleCommandSource.class)
+  public void startSetup(@NonNull CommandSource source) {
+    var setup = new RestSetup();
+    setup.applyQuestions(ANIMATION);
+
+    ANIMATION.addFinishHandler(() -> setup.handleResults(ANIMATION));
+
+    this.console.startAnimation(ANIMATION);
+  }
+
+  @CommandDescription("Displays the url of the qrcode")
+  @CommandMethod("rest qrcode")
+  @CommandPermission("cloudnet.command.rest.qrcode")
+  public void showQRCodeUrl(@NonNull CommandSource source, @Flag("appSchema") Boolean appSchema) {
+    source.sendMessage(I18n.trans("module-rest-qrcode-info"));
+    this.printExternalUrls(null, source, appSchema);
+  }
+
+  @CommandDescription("Generates a new qrcode with login information")
+  @CommandMethod("rest login <username> <password>")
+  @CommandPermission("cloudnet.command.rest.qrcode.login")
+  public void loginQRCode(@NonNull CommandSource source,
+    @NonNull @Argument(value = "username", parserName = "username") PermissionUser user,
+    @NonNull @Argument("password") String password, @Flag("appSchema") Boolean appSchema) {
+    if (!user.checkPassword(password)) {
+      source.sendMessage(I18n.trans("module-rest-qrcode-login-failed"));
+      return;
+    }
+    this.v2HttpHandlerQRCode.createSession(user);
+    source.sendMessage(I18n.trans("module-rest-qrcode-login-info"));
+    this.printExternalUrls(user.uniqueId().toString(), source, appSchema);
+  }
+
+  @Parser(name = "username")
+  public @NonNull PermissionUser username(@NonNull CommandContext<?> $, @NonNull Queue<String> input) {
+    var username = input.remove();
+    var users = CloudNetDriver.instance().permissionManagement().usersByName(username);
+    return users.stream().findFirst()
+      .orElseThrow(() -> new ArgumentNotAvailableException(I18n.trans("command-rest-username-not-found")));
+  }
+
+  private void printExternalUrls(String session, @NonNull CommandSource sender, Boolean appSchema) {
+    StringBuilder stringBuilder = new StringBuilder();
+    for (ExternalURLInformation externalUrl : ExternalURLBuilder.buildExternalURLs()) {
+      stringBuilder.append(externalUrl.toUrl());
+      stringBuilder.append("/api/v2/qrcode");
+      if (session != null) {
+        stringBuilder.append("?session=").append(session);
+      }
+      if (appSchema != null) {
+        stringBuilder.append("&appSchema=").append(true);
```

Comment:
Why are you asking for a boolean if you set it to true anyways?

Reply:
If only check if `&appSchema=` added not exactly the value